qtgui/UiApplication.cpp

```diff
@@ -0,0 +1,40 @@
+#include "UiApplication.h"
+#include "QtPixelator.h"
+#include <QGuiApplication>
+#include <QQmlApplicationEngine>
+#include "error_codes.h"
+#include "logging.h"
+
+namespace gui_mode
+{
+    int run_as_window(int argc, char* argv[], const one_bit::ArgumentParser& in_params)
+    {
+        // initialize application
+        logging::LogStream::instance().setLogLevel(logging::Level::DEBUG);
+        QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
+
+        QGuiApplication uiApp(argc, argv);
+        uiApp.setOrganizationName("starturtle");
+        uiApp.setOrganizationDomain("one-bit");
+
+        // register actual pixelation execution object
+        if (-1 == qmlRegisterType<QtPixelator>("starturtle.oneBit", 1, 0, "QtPixelator"))

```

Comment:
can that also be done with some namespaces? It would be really nice if this didn't have to be that one class without any, just because Qt is not that smart...